src/renderer/hooks/useVolume24Price.ts

```diff
@@ -8,26 +8,42 @@ import * as RxOp from 'rxjs/operators'
 
 import { useMidgardContext } from '../contexts/MidgardContext'
 import { sequenceTRD } from '../helpers/fpHelpers'
+import { triggerStream } from '../helpers/stateHelper'
 import { PriceRD } from '../services/midgard/types'
 import { AssetWithAmount } from '../types/asgardex'
 import { GetLiquidityHistoryIntervalEnum, GetSwapHistoryIntervalEnum } from '../types/generated/midgard'
 
+const { stream$: reloadHistory$, trigger: reloadHistory } = triggerStream()
+
 export const useVolume24Price = () => {
   const {
     service: {
-      pools: { poolsState$, selectedPricePool$, apiGetSwapHistory$, apiGetLiquidityHistory$ }
+      pools: { poolsState$, selectedPricePool$, apiGetSwapHistory$, apiGetLiquidityHistory$, reloadPools }
     }
   } = useMidgardContext()
 
+  const swapHistory$ = () =>
+    FP.pipe(
+      reloadHistory$,
+      RxOp.switchMap((_) => apiGetSwapHistory$({ interval: GetSwapHistoryIntervalEnum.Day, count: 1 }))
+    )
+
+  const liquidityHistory$ = () =>
+    FP.pipe(
+      reloadHistory$,
+      RxOp.switchMap((_) => apiGetLiquidityHistory$({ interval: GetLiquidityHistoryIntervalEnum.Day, count: 1 }))
+    )
+
+  const reloadVolume24Price = () => {
+    // reload of pools are needed to calculate prices properly
+    reloadPools()
+    reloadHistory()
+  }
```

Comment:
dont we want to use `useMemo/useCallback`'s ?

Reply:
I think it's not needed here, all are just lazy function without any "heavy" computations.